worlds/civ_6/__init__.py

```diff
@@ -0,0 +1,195 @@
+    """
+
+    game: str = "Civilization VI"
+    topology_present = False
+    options_dataclass = CivVIOptions
+    options: CivVIOptions
+
+    web = CivVIWeb()
+
+    item_name_to_id = {
+        item.name: item.code for item in generate_item_table().values()}
+    location_name_to_id = {
+        location.name: location.code for location in generate_flat_location_table().values()}
+
+    item_table: Dict[str, CivVIItemData] = {}
+    location_by_era: Dict[EraType, Dict[str, CivVILocationData]]
+
+    data_version = 1
+    required_client_version = (0, 4, 5)
+
+    def __init__(self, multiworld: "MultiWorld", player: int):
+        super().__init__(multiworld, player)
+        self.location_by_era = generate_era_location_table()
+
+        self.location_table: Dict[str, CivVILocationData] = {}
+        self.item_table = generate_item_table()
+
+        for _era, locations in self.location_by_era.items():
+            for _item_name, location in locations.items():
+                self.location_table[location.name] = location
+
+    def get_filler_item_name(self):
+        return get_random_filler_by_rarity(FillerItemRarity.COMMON, self.item_table).name
+
+    def create_regions(self):
+        create_regions(self, self.options, self.player)
+
+    def set_rules(self) -> None:
+        if self.options.boostsanity.value:
+            create_boost_rules(self)
+
+    def create_item(self, name: str) -> Item:
+        item: CivVIItemData = self.item_table[name]
+        classification = item.classification
+        if self.options.boostsanity.value:
+            if item.civ_name in BOOSTSANITY_PROGRESSION_ITEMS:
+                classification = ItemClassification.progression
+
+        return CivVIItem(item, self.player, classification)
+
+    def create_items(self):
+        progressive_era_item = None
+        for item_name, data in self.item_table.items():
+            # Don't add progressive items to the itempool here
+            if data.item_type == CivVICheckType.PROGRESSIVE_DISTRICT:
+                continue
+            if data.item_type == CivVICheckType.ERA:
+                # Don't add era items in this way
+                progressive_era_item = data
+                continue
+            if data.item_type == CivVICheckType.GOODY:
+                continue
+
+                # If we're using progressive districts, we need to check if we need to create a different item instead
+            item_to_create = item_name
+            if self.options.progression_style.current_key != "none":
+                item: CivVIItemData = self.item_table[item_name]
+                if item.progression_name != None:
+                    item_to_create = self.item_table[item.progression_name].name
+
+            self.multiworld.itempool += [self.create_item(
+                item_to_create)]
+
+        # Era items
+        if self.options.progression_style.current_key == "eras_and_districts":

```

Comment:
Ah nice, I didn't know this would coerce it (infer it?) like that 🔥

Reply:
Yeah, it's super convenient. Another option is to import the option class and compare to the option itself
`if self.options.progression_style == ProgressionStyle.option_eras_and_districts:`
It's not necessarily better since it's more wordy and requires an import, but it does avoid issues with misspelling strings.